x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/index/engine/FollowingEngineTests.java

```diff
@@ -450,6 +461,63 @@ public void testOptimizeSingleDocConcurrently() throws Exception {
         });
     }
 
+    public void testConcurrentUpdateOperationsWithDeletesCanAdvanceMaxSeqNoOfUpdates() throws Exception {
```

Comment:
The test here does not fail for me (without the `FollowingEngine` change), ran it 1000 times.

Can we maybe override the `advanceMaxSeqNoOfUpdatesOrDeletesOnPrimary` (and possibly other methods) in order to synchronize at the right time between the threads?